### ISSUE TITLE
'Retrying setup: None' Error

### ISSUE THREAD

**Issue by DrFate09** [marcolivierarsenault/moonraker-home-assistant#35]
![image](https://user-images.githubusercontent.com/22684695/223004392-be118707-5b70-4937-8f26-0b62a16f472d.png)

Running into the persistent error when adding a new moonraker device of "Retrying setup: None". I am unable to view any of the devices or sensors after setup.

Running HACS Moonraker version 0.3.1

Here's what I found in the log:

```
2023-03-05 21:13:27.370 ERROR (MainThread) [custom_components.moonraker] Unexpected error fetching moonraker data: 'thumbnails'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 239, in _async_refresh
    self.data = await self._async_update_data()
  File "/config/custom_components/moonraker/__init__.py", line 102, in _async_update_data
    thumbnail = await self._async_get_thumbnail(
  File "/config/custom_components/moonraker/__init__.py", line 113, in _async_get_thumbnail
    "thumbnails": gcode["thumbnails"][len(gcode["thumbnails"]) - 1][
KeyError: 'thumbnails'
2023-03-05 21:13:27.372 WARNING (MainThread) [homeassistant.config_entries] Config entry 'mainsailos' for moonraker integration not ready yet: None; Retrying in background
2023-03-05 21:13:33.151 ERROR (MainThread) [custom_components.moonraker] Unexpected error fetching moonraker data: 'thumbnails'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 239, in _async_refresh
    self.data = await self._async_update_data()
  File "/config/custom_components/moonraker/__init__.py", line 102, in _async_update_data
    thumbnail = await self._async_get_thumbnail(
  File "/config/custom_components/moonraker/__init__.py", line 113, in _async_get_thumbnail
    "thumbnails": gcode["thumbnails"][len(gcode["thumbnails"]) - 1][
KeyError: 'thumbnails'
```

**Comment by marcolivierarsenault** [comment]
can you try to completely remove the device and add it back. 
https://github.com/marcolivierarsenault/moonraker-home-assistant/wiki#full-reload-of-the-integration

**Comment by marcolivierarsenault** [comment]
Did it worked in a previous version

**Comment by DrFate09** [comment]
I followed the instructions.

Uninstalled; rebooted
Installed; rebooted
Added IP of printer and still seeing the same error.

Anything else I can try?

It was working in version 0.2.0

**Comment by marcolivierarsenault** [comment]
Can you send me the output if you open your browser with

192.168.1.1/server/files/metadata

(change IP for IP of your moonraker)

**Comment by DrFate09** [comment]
`{"error": {"code": 400, "message": "No data for argument: filename", "traceback": "Traceback (most recent call last):\n\n  File \"/home/pi/moonraker/moonraker/app.py\", line 693, in _process_http_request\n    result = await self._do_request(args, conn)\n\n  File \"/home/pi/moonraker/moonraker/app.py\", line 672, in _do_local_request\n    user=self.current_user))\n\n  File \"/home/pi/moonraker/moonraker/components/file_manager/file_manager.py\", line 377, in _handle_metadata_request\n    requested_file: str = web_request.get_str('filename')\n\n  File \"/home/pi/moonraker/moonraker/common.py\", line 327, in get_str\n    return self._get_converted_arg(key, default, str)\n\n  File \"/home/pi/moonraker/moonraker/common.py\", line 295, in _get_converted_arg\n    raise ServerError(f\"No data for argument: {key}\")\n\nmoonraker.utils.ServerError: No data for argument: filename\n\n\nThe above exception was the direct cause of the following exception:\n\n\nTraceback (most recent call last):\n\n  File \"/home/pi/moonraker-env/lib/python3.7/site-packages/tornado/web.py\", line 1713, in _execute\n    result = await result\n\n  File \"/home/pi/moonraker/moonraker/app.py\", line 656, in get\n    await self._process_http_request()\n\n  File \"/home/pi/moonraker/moonraker/app.py\", line 696, in _process_http_request\n    e.status_code, reason=str(e)) from e\n\ntornado.web.HTTPError: HTTP 400: No data for argument: filename\n"}}`

**Comment by DrFate09** [comment]
@marcolivierarsenault Thank fixed the issue for me.

Log: [home-assistant_moonraker_2023-03-06T03-34-12.034Z.log](https://github.com/marcolivierarsenault/moonraker-home-assistant/files/10894145/home-assistant_moonraker_2023-03-06T03-34-12.034Z.log)

**Comment by marcolivierarsenault** [comment]
Great Solved in 0.3.2
